pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1397,27 +1397,48 @@ private static int integerPart(String dirtyString) {
     return Integer.parseInt(dirtyString.substring(start, end));
   }
 
+  // Validation check for integer values that need to be positive
+  // mainly added to check resultSetType, resultSetConcurrency and resultSetHoldability
+  private static boolean isPositive(int value) {
+    if (value < 0) {
+      return false;
+    }
+    return true;
+  }
+
   @Override
   public Statement createStatement(int resultSetType, int resultSetConcurrency,
       int resultSetHoldability) throws SQLException {
     checkClosed();
-    return new PgStatement(this, resultSetType, resultSetConcurrency, resultSetHoldability);
+    if (isPositive(resultSetType) && isPositive(resultSetConcurrency) && isPositive(resultSetHoldability)) {
+      return new PgStatement(this, resultSetType, resultSetConcurrency, resultSetHoldability);
+    } else {
+      throw new IllegalArgumentException("Value must be positive");
+    }
   }
 
   @Override
   public PreparedStatement prepareStatement(String sql, int resultSetType, int resultSetConcurrency,
       int resultSetHoldability) throws SQLException {
     checkClosed();
-    return new PgPreparedStatement(this, sql, resultSetType, resultSetConcurrency,
-        resultSetHoldability);
+    if (isPositive(resultSetType) && isPositive(resultSetConcurrency) && isPositive(resultSetHoldability)) {
+      return new PgPreparedStatement(this, sql, resultSetType, resultSetConcurrency,
+          resultSetHoldability);
+    } else {
+      throw new IllegalArgumentException("Value must be positive");
```

Comment:
Which value should be "positive"?
I believe we should throw `SQLException` rather than `IllegalArgumentException`.
We should support localization for the exception, and the error message should better include expected and the actual values

Reply:
can take this up along with resultSet Validation mentioned above